Private/Find-ESC4.ps1

```diff
@@ -115,14 +121,16 @@ Set-ACL -Path `'AD:$($_.DistinguishedName)`' -AclObject `$ACL
         }
 
         foreach ($entry in $_.nTSecurityDescriptor.Access) {
-            $Principal = New-Object System.Security.Principal.NTAccount($entry.IdentityReference)
-            if ($Principal -match '^(S-1|O:)') {
-                $SID = $Principal
-            } else {
-                $SID = ($Principal.Translate([System.Security.Principal.SecurityIdentifier])).Value
+            if ($_.Name -ne '' -and $null -ne $_.Name) {
```

Comment:
This is a good place to avoid errors and simplify by using `if ( [string]::IsNullOrEmpty($Name) ) { }`.